src/window.c

```diff
@@ -954,6 +954,95 @@ GLFWAPI void glfwSetWindowAttrib(GLFWwindow* handle, int attrib, int value)
     _glfwInputError(GLFW_INVALID_ENUM, "Invalid window attribute 0x%08X", attrib);
 }
 
+typedef struct _rect { int x, y; int w, h; } _rect;
+
+#define RECT_INTERSECTS(ra, rb) \
+  ((ra)->x <= ((rb)->x + (rb)->w) && ((ra)->x + (ra)->w) >= (rb)->x && \
+  (ra)->y <= ((rb)->y + (rb)->h) && ((ra)->y + (ra)->h) >= (rb)->y)
+
+static _rect _get_intersection(_rect* ra, _rect* rb)
```

Comment:
I think `_` prefix is only used on internal functions with `glfw` in their name. All helper functions do not appear to have this prefix.